src/ripple/json/MultivarJson.h

```diff
@@ -48,11 +48,50 @@ struct MultivarJson
         for (auto& a : this->val)
             a[key] = v;
     }
+
+    // Intentionally not using class enum here, MultivarJson is scope enough
+    enum IsMemberResult : int { none = 0, some, all };
+
+    [[nodiscard]] IsMemberResult
+    isMember(const char* key) const
+    {
+        int count = 0;
+        for (auto& a : this->val)
+            if (a.isMember(key))
+                count += 1;
+
+        return (count == 0 ? none : (count < size ? some : all));
+    }
 };
 
 // Wrapper for Json for all supported API versions.
 using MultiApiJson = MultivarJson<2>;
 
+/*
+
+NOTE:
+
+If a future API version change adds another possible format, change the size of
+`MultiApiJson`, and update `apiVersionSelector()` to return the appropriate
+selection value for the new `apiVersion` and higher.
+
+e.g. There are 2 formats now, the first, for version one, the second for
+versions > 1. Hypothetically, if API version 4 adds a new format, `MultiApiJson`
+would be MultivarJson<3>, and `apiVersionSelector` would return
+`static_cast<std::size_t>(apiVersion < 2 ? 0u : (apiVersion < 4 ? 1u : 2u))`
+
+NOTE:
+
+The more different JSON formats we support, the more CPU cycles we need to
+pre-build JSON for different API versions e.g. when publishing streams to
+`subscribe` clients. Hence it is desirable to keep MultiApiJson small and
+instead fully deprecate and remove support for old API versions. For example, if
+we removed support for API version 1 and added a different format for API
+version 3, the `apiVersionSelector` would change to
+`static_cast<std::size_t>(apiVersion > 2)`
```

Comment:
No need to change code or comments, just something to think about.

Another possible solution is lazy loading, where we build a "base" version of the object, then provide a function to make the necessary per-apiVersion changes. When a new version is `select`ed, it'll copy and modify base object and store the result in the array.

Reply:
Yeah I want to leave some guide there, so ppl do not have idea that we must always maintain some specific version for sake of this array.